mittab/libs/tab_logic.py

```diff
@@ -806,53 +851,67 @@ def debater_abnormal_round_ranks(debater, round_number):
     Uses average ranks
     """
     team = deb_team(debater)
-    had_bye = Bye.objects.filter(round_number=round_number,
-                                    bye_team=team)
-    had_noshow = NoShow.objects.filter(round_number=round_number,
-                                        no_show_team=team)
+    had_bye = Bye.objects.filter(round_number=round_number, bye_team=team)
+    had_noshow = NoShow.objects.filter(round_number=round_number, no_show_team=team)
+
     if had_bye or (had_noshow and had_noshow.first().lenient_late):
         return avg_deb_ranks(debater)
+
     elif had_noshow:
         return MAXIMUM_DEBATER_RANKS
 
 
 def single_adjusted_ranks_deb(debater):
     debater_ranks = ranks_for_debater(debater)
-    debater_ranks.sort()
-    return sum(debater_ranks[1:-1])
+    return sum(_single_adjust(sorted(debater_ranks)))
 
 def double_adjusted_ranks_deb(debater):
     debater_ranks = ranks_for_debater(debater)
-    debater_ranks.sort()
-    return sum(debater_ranks[2:-2])
+    return sum(_double_adjust(sorted(debater_ranks)))
 
 @cache()
 def tot_ranks_deb(debater, average_ironmen=True):
     debater_ranks = ranks_for_debater(debater, average_ironmen=average_ironmen)
     return sum(debater_ranks)
 
+
 def deb_team(d):
     try:
-        return d.team_set.all()[0]
-    except:
+        return d.team_set.first()
```

Comment:
Interestingly, I'm finding that changing from `all()[0] to `first()` actually makes things noticeably slower (adds about 1 second). Maybe it's because nobody is every really on more than 1 team, and `first()` adds some more sql logic that makes the query slower than getting all of them?

Reply:
Analytically, I would imagine that wouldn't be the case, since the former should produce an SQL query with more runtime than the latter. But empirics don't lie. Thinking about how that could be the case, perhaps it has to do with caching or prefetch changes?